# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../metamail/20_mm-2.7-uudecode.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# Copyright (C) 1998 - 2003 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- mm2.7/src/bin/sun-audio-file.ewt	Fri Oct 24 09:41:11 1997
+++ mm2.7/src/bin/sun-audio-file	Fri Oct 24 09:52:05 1997
@@ -6,7 +6,13 @@
     set METAMAIL_TMPDIR=/tmp
 endif
 
-cd ${METAMAIL_TMPDIR}
-uudecode < $1
-audiotool audio-file
-rm -f audio-file $1
+set fn=`mktemp $METAMAIL_TMPDIR/audio.XXXXXX`
+test -n "$fn"
+if $? then
+	exit 1
+endif
+
+uudecode -o $fn < $1
+
+audiotool $fn
+rm -f $fn $1
--- mm2.7/src/bin/sun-message.csh.ewt	Fri Oct 24 09:44:07 1997
+++ mm2.7/src/bin/sun-message.csh	Fri Oct 24 09:52:30 1997
@@ -9,7 +9,13 @@
 cd /tmp/decode.$$
 
 if ($2 == "uuencode") then
-    uudecode $1
+    set fn=`mktemp $METAMAIL_TMPDIR/audio.XXXXXX`
+    test -n "$fn"
+    if $? then
+	    exit 1
+    endif
+
+    uudecode -o $fn $1
 
     echo "The following file was uudecoded:"
     echo ""
@@ -17,7 +23,6 @@
     set defans = "1"
     while (1)
 	ls -l
-	set fn = *
 
 	echo ""
 	echo "Please choose one:"
